sdk/python/tests/components/test_python_op.py

```diff
@@ -184,6 +184,81 @@ def add_multiply_two_numbers(a: float, b: float) -> NamedTuple('DummyName', [('s
 
         self.helper_test_2_in_2_out_component_using_local_call(func, op, output_names=['sum', 'product'])
 
+    def test_extract_component_interface(self):
+        from typing import NamedTuple
+        def my_func(
+            required_param,
+            int_param: int = 42,
+            float_param : float = 3.14,
+            str_param : str = 'string',
+            bool_param : bool = True,
+            none_param = None,
+            custom_type_param: 'Custom type' = None,
+            ) -> NamedTuple('DummyName', [
+                #('required_param',), # All typing.NamedTuple fields must have types
+                ('int_param', int),
+                ('float_param', float),
+                ('str_param', str),
+                ('bool_param', bool),
+                #('custom_type_param', 'Custom type'), #SyntaxError: Forward reference must be an expression -- got 'Custom type'
```

Comment:
nit: Did you leave the comment intentionally?

Reply:
Yes. This is a reminder of an issue with type names - NamedTuple fails on names with spaces.